blockmanager.go

```diff
@@ -3043,4 +3097,177 @@ func (l *lightHeaderCtx) RelativeAncestorCtx(
 	return newLightHeaderCtx(
 		ancestorHeight, ancestor, l.store, l.headerList,
 	)
+
+}
+
+func (b *blockManager) sideLoadHeaders() {
+	// If sideloading is not enabled on this chain
+	// return quickly.
+	if !b.cfg.sideLoad.Enabled {
+		return
+	}
+
+	// If headers contained in the side load source are for a different chain network return
+	// immediately.
+	if b.sideLoadReader.HeadersChain() != b.cfg.ChainParams.Net {
+		log.Error("headers from side load file are of network %v "+
+			"and so incompatible with neutrino's current bitcoin network "+
+			"-- skipping side loading", b.sideLoadReader.HeadersChain())
+
+		return
+	}
+	log.Infof("Side loading headers from %v to %v", b.headerTip, b.sideLoadReader.EndHeight())
+
+	// Set headerTip to enable reader supply header, node needs
+	err := b.sideLoadReader.SetHeight(int64(b.headerTip))
+	if err != nil {
+		log.Errorf("error while setting height for sideload--- skipping sideloading: "+
+			"%v", err)
+
+		return
+	}
+
+	// Start batch writer to write headers into DB at specified batch size and ticker duration.
+	b.blockHeaderBatchWriter.Start()
+	defer b.blockHeaderBatchWriter.Stop()
+	for {
+		select {
+		case <-b.quit:
+			return
+		default:
+			//Request header
+			header, headerErr := b.sideLoadReader.NextHeader()
+			// If any error occurs while fetching headers that does not indicate
+			// an end of file, return immediately.
+			if headerErr != nil {
+
+				if headerErr == chainDataLoader.ErrEndOfFile {
+					log.Infof("Successfully completed sideloading")
+					return
+				}
+
+				log.Errorf("error while fetching headers -- skipping sideloading %v", err)
+				return
+			}
+
+			var (
+				node     *headerlist.Node
+				prevNode *headerlist.Node
+			)
+			// Ensure there is a previous header to compare against.
+			prevNodeEl := b.headerList.Back()
```

Comment:
Okay thanks.

Reply:
There is actually another variable called `prevNode`, I copy `prevNodeEl` to `prevNode`, so that the value of the nodes being pushed to the header list would not be overwritten during the iteration.